compiler/luci/pass/src/ResolveCustomOpMatMulPass.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/ResolveCustomOpMatMulPass.h"
+
+#include "flatbuffers/flexbuffers.h"
+#include <loco/IR/DataTypeTraits.h>
+
+#include <luci/IR/CircleNodes.h>
+
+#include <loco.h>
+#include <oops/InternalExn.h>
+#include <loco/Service/ShapeInference.h>
+#include <loco/Service/TypeInference.h>
+
+namespace
+{
+
+template <typename T>
+luci::CircleConst *create_const_node(loco::Graph *g, const loco::DataType dtype,
+                                     const std::vector<uint32_t> &shape,
+                                     const std::vector<T> &values)
+{
+  auto node = g->nodes()->create<luci::CircleConst>();
+  node->dtype(dtype);
+  node->rank(shape.size());
+
+  uint32_t size = 1;
+  for (uint32_t i = 0; i < shape.size(); ++i)
+  {
+    node->dim(i) = shape.at(i);
+    size *= shape.at(i);
+  }
+
+  switch (dtype)
+  {
+    case loco::DataType::S32:
+      node->size<loco::DataType::S32>(size);
+      for (uint32_t i = 0; i < values.size(); ++i)
+        node->at<loco::DataType::S32>(i) = values[i];
+      break;
+    case loco::DataType::S64:
+      node->size<loco::DataType::S64>(size);
+      for (uint32_t i = 0; i < values.size(); ++i)
+        node->at<loco::DataType::S64>(i) = values[i];
+      break;
+    case loco::DataType::FLOAT32:
+      node->size<loco::DataType::FLOAT32>(size);
+      for (uint32_t i = 0; i < values.size(); ++i)
+        node->at<loco::DataType::FLOAT32>(i) = values[i];
+      break;
+    default:
+      return nullptr;
```

Comment:
I didn't check the documents but if `create_const_node` only supports S32, S64 and FLOAT32 and nothing else, then it would be better to throw there to be clear.

Reply:
Done